src/main/webapp/src/App.js

```diff
@@ -23,53 +27,144 @@ class App extends React.Component {
+            top: 70,
+            message: `The ${name} successfully added to cart!`,
+            duration: 1.5,
+            icon: <CheckCircleOutlined style={{color: '#108ee9', fontSize: 30}}/>,
+        });
+    }
+
+
+    addToCart = (id, name) => {
+        const userId = this.getUserId();
+        userId ?
+            axios.post("/api/public/basket", {id: id, userId: userId}).then(
+                response => {
+                    this.getBasket(response, name);
+                }
+            )
+            :
+            axios.post("/api/public/basket/unregistered", {id: id})
+                .then(response => {
+                        this.getBasket(response, name);
+                    }
+                );
     }
 
     clearBasket = () => {
-        axios.delete("/api/basket/clear").then(response => {
-            this.setState({basket: undefined});
+        axios.delete("/api/public/basket/clear").then(response => {
+            this.setState({basket: null});
         });
     }
 
     removeItem = (id) => {
-        axios.delete(`/api/basket/removeitem/${id}`).then(response => {
-                axios.get('/api/basket').then(response => {
-                    const basket = response.data;
-                    this.setState({basket});
-                });
+        const userId = this.getUserId();
+        axios.delete(`/api/public/basket/removeitem/${id}`).then(response => {
+                userId ?
+                    axios.get('/api/public/basket', {params: {userId: userId}}).then(response => {
+                        const basket = response.data;
+                        this.setState({basket});
+                    })
+                    :
+                    axios.get('/api/public/basket/unregistered').then(response => {
+                        const basket = response.data;
+                        this.setState({basket});
+                    })
+            }
+        );
+    }
+
+    logOut = () => {
+        axios.post("/api/public/auth/logout").then(response => {
+            notification.open({
+                top: 70,
+                message: `${this.state.currentUser.username}, you have successfully logged out of your account!`,
+                duration: 2.5,
+                icon: <LogoutOutlined style={{color: '#108ee9', fontSize: 30}}/>,
+            })
+            this.setState({user: null});
+            this.setState({currentUser: null});
+            Cookie.deleteCookie("user");
+        });
+        axios.delete("/api/public/basket/logout").then(response => {
+                this.setState({basket: null});
             }
         );
     }
 
+    submitForm = (username, password) => {
+        axios.post("/api/public/auth/login", {username: username, password: password}).then(response => {
+            this.setState({currentUser: response.data});
+            if (this.state.currentUser) {
+                Cookie.setCookie("user", JSON.stringify(this.state.currentUser));
+            }
```

Comment:
do you actually need this info in cookies? why do you need userId on client?

Reply:
+

Reply:
+